scipy-stubs/signal/_spectral_py.pyi

```diff
@@ -1,97 +1,209 @@
-from typing import Literal
+from collections.abc import Callable
+from typing import Any, Literal, TypeAlias, overload
+from typing_extensions import Unpack
 
-from scipy._typing import Untyped, UntypedCallable
+import numpy as np
+import numpy.typing as npt
+import optype as op
+from numpy._typing import _ArrayLikeComplex_co, _ArrayLikeFloat_co
```

Comment:
It's a bit awkward, but `_ArrayLikeComplex_co` is equivalent to `_ArrayLikeNumber_co`, since `np.integer[Any] | np.floating[Any] | np.complexfloating[Any, Any]` is equivalent to `np.number[Any]`. 

So I usually prefer the more concise `_ArrayLikeNumber_co`, because that could help with type-checker performance, and makes the potential error messages of type-checkers ~more readable~ less unreadable.